### PR TITLE
DNM Archive custom artifacts after job complete

### PR DESCRIPTION
**Pull request by kshtsk**
Example:

    override:
        archive_artifacts:
        - '/etc/zypp/repos.d'
        - '/etc/resolv.conf'
        - name: 'zypper-repos'
          exec: 'sudo zypper lr'
        - name: 'nslookup-suse.com'
          exec: 'nslookup suse.com 2>&1'

### CODE REVIEW

File: teuthology/task/install/__init__.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ def remove_packages(ctx, config, pkgs):
     with parallel() as p:
         for remote in ctx.cluster.remotes.iterkeys():
             system_type = teuthology.get_system_type(remote)
+            if remote.os.name in ['opensuse', 'sle']:
+                remote.run(args='zypper lr')
```
Comment (smithfarm) [inline teuthology/task/install/__init__.py]: I find `zypper lr -upEP` to be more informative.

File: teuthology/task/internal/__init__.py [hunk 2]
```diff
@@ -459,6 +459,78 @@ def coredump(ctx, config):
...
+@contextlib.contextmanager
+def archive_artifacts(ctx, config):
+    """
+    Stash a artifacts.
+    override:
+        archive_artifacts:
+        - '/etc/zypp/repos.d'
+        - '/etc/resolv.conf'
+        - name: 'zypper-repos'
+          exec: 'sudo zypper lr'
```
Comment (smithfarm) [inline teuthology/task/internal/__init__.py]: `sudo zypper lr -upEP`

File: teuthology/task/install/__init__.py (hunk 1 above)
Comment (kshtsk) [reply teuthology/task/install/__init__.py]: Oops, those two lines should not be present here, forgot to remove it after debugging.

File: teuthology/task/internal/__init__.py (hunk 2 above)
Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: Well, this is just an example, the idea is that you can specify any command sequence supposed to be executed to capture output in the accompanying yaml file.

**Comment by kshtsk** [comment]
@susebot run teuthology for suse:tier1

**Comment by kshtsk** [comment]
@susebot run teuthology for suse:tier0

**Comment by kshtsk** [comment]
Closing it for now, maybe it can be reworked later

**Pull request closed without merging**